src/meta/meta_service.h

```diff
@@ -195,9 +213,29 @@ class meta_service : public serverlet<meta_service>
     // if return -1 and `forward_address' != nullptr, then return leader by `forward_address'.
     int check_leader(dsn::message_ex *req, dsn::rpc_address *forward_address);
     template <typename TRpcHolder>
-    int check_leader(TRpcHolder rpc, /*out*/ rpc_address *forward_address);
-    template <typename TRpcHolder>
-    bool check_status(TRpcHolder rpc, /*out*/ rpc_address *forward_address = nullptr);
+    int check_leader(TRpcHolder rpc, rpc_address *forward_address)
```

Comment:
What did this modification fix? Why move the implementation to the header file?

Reply:
If we don't move this to header file, the link will produce error. It seems it can't do the template instantiation work if we don't move it to header file.

Reply:
I moved the implementation of `check_status` and `check_leader` out of class `meta_service`, but they are still being put in the header file.